src/fsharp/NameResolution.fs

```diff
@@ -1538,13 +1538,16 @@ type TcResultsSinkImpl(g, ?source: string) =
     let formatStringCheckContext =
         lazy
             source |> Option.map (fun source ->
-                let source = source.Replace("\r\n", "\n").Replace("\r", "\n")
                 let positions =
-                    source.Split('\n')
-                    |> Seq.map (fun s -> String.length s + 1)
-                    |> Seq.scan (+) 0
-                    |> Seq.toArray
-                { NormalizedSource = source 
+                    [|
+                        let mutable pos = 0
+                        yield pos
+                        for c in source do
+                            if c = '\r' then ()
```

Comment:
ok I now believe this introduces a bug on files that are `\r` delimited (IIRC that's mac style).
you should probably introduce a mutable bool to detect these cases and handle one char later